apps/dashboard/src/pages/integrations/components/create-integration-sidebar.tsx

```diff
@@ -0,0 +1,124 @@
+import { ChannelTypeEnum, providers as novuProviders } from '@novu/shared';
+import { useCreateIntegration } from '@/hooks/use-create-integration';
+import { useIntegrationList } from './hooks/use-integration-list';
+import { useSidebarNavigationManager } from './hooks/use-sidebar-navigation-manager';
+import { IntegrationSheet } from './integration-sheet';
+import { ChannelTabs } from './channel-tabs';
+import { IntegrationConfiguration } from './integration-configuration';
+import { Button } from '../../../components/primitives/button';
+import { handleIntegrationError } from './utils/handle-integration-error';
+import { useSetPrimaryIntegration } from '../../../hooks/use-set-primary-integration';
+import { SelectPrimaryIntegrationModal } from './modals/select-primary-integration-modal';
+import { IntegrationFormData } from '../types';
+import { useIntegrationPrimaryModal } from './hooks/use-integration-primary-modal';
+import { useFetchIntegrations } from '@/hooks/use-fetch-integrations';
+
+export type CreateIntegrationSidebarProps = {
+  isOpened: boolean;
+  onClose: () => void;
+  scrollToChannel?: ChannelTypeEnum;
+};
+
+export function CreateIntegrationSidebar({ isOpened, onClose }: CreateIntegrationSidebarProps) {
+  const providers = novuProviders;
+  const { mutateAsync: createIntegration, isPending } = useCreateIntegration();
+  const { mutateAsync: setPrimaryIntegration, isPending: isSettingPrimary } = useSetPrimaryIntegration();
+  const { integrations } = useFetchIntegrations();
+
+  const { selectedIntegration, step, searchQuery, onIntegrationSelect, onBack } = useSidebarNavigationManager({
+    isOpened,
+  });
+
+  const { integrationsByChannel } = useIntegrationList(providers, searchQuery);
+  const provider = providers?.find((p) => p.id === selectedIntegration);
+  const {
+    isPrimaryModalOpen,
+    setIsPrimaryModalOpen,
+    pendingData,
+    handleSubmitWithPrimaryCheck,
+    handlePrimaryConfirm,
+    existingPrimaryIntegration,
+    isChannelSupportPrimary,
+  } = useIntegrationPrimaryModal({
+    onSubmit,
+    integrations,
+    channel: provider?.channel,
+    mode: 'create',
+  });
+  async function onSubmit(data: IntegrationFormData) {
+    if (!provider) return;
+
+    try {
+      const integration = await createIntegration({
+        providerId: provider.id,
+        channel: provider.channel,
+        credentials: data.credentials,
+        name: data.name,
+        identifier: data.identifier,
+        active: data.active,
+        _environmentId: data.environmentId,
+      });
+
+      if (data.primary && isChannelSupportPrimary && data.active) {
+        await setPrimaryIntegration({ integrationId: integration.data._id });
+      }
+
+      onClose();
+    } catch (error: any) {
+      handleIntegrationError(error, 'create');
+    }
+  }
+
```

Comment:
It's super hard to understand what's going on here, I suggest clearing it a bit. For example `onSubmit` is used by the `useIntegrationPrimaryModal` hook but is defined here.

Reply:
Let me try to refactor, the onSubmit is passed to the hack because we use the hook for both update and create for reusability. the onSubmit is just an action handler